### PR TITLE
added code summary table

### PR DESCRIPTION
**Pull request by ps48**
Signed-off-by: Shenoy Pratik <sgguruda@amazon.com>

### Description
The code summary table gives the latest information on current status of builds, tests, and issues.

### Issues Resolved
Used Github badges to add summary in the repository.
Note: IT badges are currently dummy. 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by davidcui1225 — DISMISSED** [review]

**Review by davidcui1225 — APPROVED** [review]

**Comment by davidcui1225** [comment]
LGTM - can you merge from upstream main to fix the failing CIs

File: README.md [hunk 1]
```diff
@@ -20,6 +16,75 @@
...
+| Test and build               | [![Observability OpenSearch Build CI][opensearch-build-badge]][opensearch-build-link]                                                        |
+| Code coverage                | [![codecov][opensearch-codecov-badge]][codecov-link]                                                                                         |
+| Distribution build tests     | [![OpenSearch IT tests][opensearch-it-badge]][opensearch-it-link] [![OpenSearch IT code][opensearch-it-code-badge]][opensearch-it-code-link] |
+| Backward compatibility tests | [![BWC tests][bwc-tests-badge]][bwc-tests-link]                                                                                              |
+
+### OpenSearch Dashboards-Plugin
+
+|                          |                                                                                                                    |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------ |
+| Test and build           | [![Observability Dashboards CI][dashboard-build-badge]][dashboard-build-link]                                      |
```
Comment (zhongnansu) [inline README.md:34]: please change the naming to dashboars-reports or reports-scheduler

File: README.md [hunk 2]
```diff
@@ -20,6 +16,75 @@
...
+[opensearch-codecov-badge]: https://codecov.io/gh/opensearch-project/dashboards-reports/branch/main/graphs/badge.svg?flag=reports-scheduler
+[codecov-link]: https://codecov.io/gh/opensearch-project/dashboards-reports
+[cypress-test-badge]: https://img.shields.io/badge/Cypress%20tests-in%20progress-yellow
+[cypress-test-link]: https://github.com/opensearch-project/opensearch-build/issues/1124
+[cypress-code-badge]: https://img.shields.io/badge/Cypress%20code-blue
+[cypress-code-link]: https://github.com/opensearch-project/dashboards-reports/tree/main/dashboards-reports/.cypress/integration
+[opensearch-it-badge]: https://img.shields.io/badge/OpenSearch%20Plugin%20IT%20tests-in%20progress-yellow
+[opensearch-it-link]: https://github.com/opensearch-project/opensearch-build/issues/1124
+[opensearch-it-code-badge]: https://img.shields.io/badge/OpenSearch%20IT%20code-blue
+[opensearch-it-code-link]: https://github.com/opensearch-project/dashboards-reports/blob/main/reports-scheduler/src/test/kotlin/org/opensearch/observability/ObservabilityPluginIT.kt
```
Comment (zhongnansu) [inline README.md:72]: same here

**Review by zhongnansu — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -20,6 +16,75 @@
...
+[bwc-tests-badge]: https://img.shields.io/badge/BWC%20tests-in%20progress-yellow
+[bwc-tests-link]: https://github.com/opensearch-project/dashboards-reports/issues/240
```
Comment (zhongnansu) [inline README.md:74]: you may change this, since the PR has just been merged in
https://github.com/opensearch-project/dashboards-reports/pull/244

Comment (ps48) [reply README.md:74]: Should I link this to the #244 files for now? Once, the bwc tests start running on Jenkins' distribution builds we can link to it.

Comment (zhongnansu) [reply README.md]: yeah linking to the files for now makes sense to me.

**Review by zhongnansu — APPROVED** [review]

**Review by davidcui1225 — APPROVED** [review]

**Pull request merged**